pkg/controller/workload/cache/workload_cache.go

```diff
@@ -70,26 +70,58 @@ func composeNetworkAddress(network string, addr netip.Addr) NetworkAddress {
 	}
 }
 
-func (w *cache) AddWorkload(workload *workloadapi.Workload) {
+func (w *cache) getUniqueServicesOnLeftWorkload(workload1, workload2 *workloadapi.Workload) []string {
+	var diff []string
+	if workload1 == nil {
+		return diff
+	}
+
+	for key := range workload1.Services {
+		if workload2 == nil {
+			diff = append(diff, key)
+			continue
+		}
+		if _, exist := workload2.Services[key]; !exist {
+			diff = append(diff, key)
+		}
+	}
+	return diff
+}
+
+func (w *cache) compareWorkloadServices(workload1, workload2 *workloadapi.Workload) ([]string, []string) {
```

Comment:
This is not efficient either, in both getUniqueServicesOnLeftWorkload, it will iterate over both workloads